front/tests/pages/stdcm-page-model.ts

```diff
@@ -413,24 +505,30 @@ class STDCMPage {
         await this.getViaType(viaNumber).selectOption('driverSwitch');
         await expect(this.getViaStopTime(viaNumber)).toHaveValue('3');
         await this.getViaStopTime(viaNumber).fill('2');
-
         await expect(warning).toBeVisible();
         expect(await warning.textContent()).toEqual(translations.trainPath.warningMinStopTime);
-
         await this.getViaStopTime(viaNumber).fill('4');
         await expect(warning).not.toBeVisible();
         break;
 
       default:
-        throw new Error(`Unsupported viaSearch value: ${viaSearch}`);
+        throw new Error(`Unsupported viaSearch value: ${ciSearchText}`);
     }
   }
 
   // Launches the simulation and checks if simulation-related elements are visible
   async launchSimulation() {
+    await this.launchSimulationButton.waitFor();
     await expect(this.launchSimulationButton).toBeEnabled();
-    await this.launchSimulationButton.click();
-    await expect(this.simulationList).toBeVisible();
+    await this.page.waitForTimeout(500); // Small buffer
```

Comment:
Can we avoid introducing more of these?

Reply:
I know using `waitForTimeout` is discouraged, but the application often has latency, especially in local tests. I updated the function to work in CI, but it's hard to confirm locally

Reply:
If we need a `waitForTimeout()`, that means that we're missing a condition somewhere. In other words, instead of `waitForTimeout()`, we should wait for the app to update itself in another way (an element becomes visible, etc).